src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventPipeEventProvider.cs

```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if FEATURE_PERFTRACING
```

Comment:
I think the existing setup prefers to exclude/include files .projitem file over if-defs

Reply:
Yes, wasn't sure the path forward on this one, there are also some build errors due to this, so I will adjust. Should we change that for all the sources using FEATURE_PERFTRACING, do exclude/include in projitem?

Reply:
Turns out that defines are still needed since files gets included by net472 csproj files using *.cs pattern, so not using the include/exclude patterns added to .projitems at the moment.

Reply:
It would look better to switch the net472 csproj file to use explicit file list.

Reply:
We could follow current pattern to remove files that shouldn't be included or did you think of actually list each specific file and not rely on wildcards at all in runtime\src\libraries\Microsoft.Diagnostics.Tracing.EventSource.Redist\src\Microsoft.Diagnostics.Tracing.EventSource.Redist.csproj, so converting this:

<Compile Include="..\..\System.Private.CoreLib\src\System\Diagnostics\Tracing\*.cs" />
    <Compile Remove="..\..\System.Private.CoreLib\src\System\Diagnostics\Tracing\FrameworkEventSource.cs" />
    <Compile Include="..\..\System.Private.CoreLib\src\System\Diagnostics\Tracing\TraceLogging\*.cs" />

to list of <Compile include=""> statements?

Reply:
I would list each specific file. The `Microsoft.Diagnostics.Tracing.EventSource.Redist` is unlikely to evolve much. I would optimize for it not changing, and not getting impacted by CoreLib changes.